src/printer.js

```diff
@@ -2342,6 +2354,7 @@ function typeIsFunction(type) {
 
 function printExportDeclaration(path, options, print) {
   const decl = path.getValue();
+  const semi =  options.noSemi ? "" : ";"
```

Comment:
This looks very weird indeed for the removal of newlines. May be related to https://github.com/prettier/prettier/pull/1136